tests/expressions/test_null_safe_equals.py

```diff
@@ -0,0 +1,483 @@
+
+
+@pytest.mark.parametrize(
+    "type_name,test_value,test_data,expected_values",
+    [
+        ("int", 10, [(1, 10), (2, None), (3, 20)], [True, False, False]),
+        ("string", "hello", [(1, "hello"), (2, None), (3, "world")], [True, False, False]),
+        ("boolean", True, [(1, True), (2, None), (3, False)], [True, False, False]),
+        ("float", 1.5, [(1, 1.5), (2, None), (3, 2.5)], [True, False, False]),
+        ("binary", b"hello", [(1, b"hello"), (2, None), (3, b"world")], [True, False, False]),
+        ("fixed_size_binary", b"aaa", [(1, b"aaa"), (2, None), (3, b"bbb")], [True, False, False]),
+    ],
+)
+def test_null_safe_equals_types(type_name, test_value, test_data, expected_values):
+    """Test null-safe equality with different data types."""
+    import pyarrow as pa
+
+    # Create a table with the test data
+    if type_name == "fixed_size_binary":
+        # Convert to PyArrow array for fixed size binary
+        value_array = pa.array([x[1] for x in test_data], type=pa.binary(3))
+        table = MicroPartition.from_pydict(
+            {
+                "id": [x[0] for x in test_data],
+                "value": value_array.to_pylist(),
+            }
+        )
+    else:
+        table = MicroPartition.from_pydict(
+            {
+                "id": [x[0] for x in test_data],
+                "value": [x[1] for x in test_data],
+            }
+        )
+
+    # Apply the null-safe equals operation
+    result = table.eval_expression_list([col("value").eq_null_safe(lit(test_value))])
+    result_values = result.get_column("value").to_pylist()
+
+    # Print full comparison for debugging
+    print("\nFull comparison:")
+    print("Result values:")
+    for val, res in zip([x[1] for x in test_data], result_values):
+        print(f"  value={val}, eq_value={res}")
+    print("\nExpected values:")
+    for val, exp in zip([x[1] for x in test_data], expected_values):
+        print(f"  value={val}, expected={exp}")
+
+    # Verify results
+    assert result_values == expected_values, f"Failed for {type_name} comparison"
+
+
+@pytest.mark.parametrize(
+    "type_name,test_value,test_data,expected_values",
+    [
+        ("int", 10, [(1, 10), (2, None), (3, 20)], [True, False, False]),
+        ("string", "hello", [(1, "hello"), (2, None), (3, "world")], [True, False, False]),
+        ("boolean", True, [(1, True), (2, None), (3, False)], [True, False, False]),
+        ("float", 1.5, [(1, 1.5), (2, None), (3, 2.5)], [True, False, False]),
+        ("binary", b"hello", [(1, b"hello"), (2, None), (3, b"world")], [True, False, False]),
+        ("fixed_size_binary", b"aaa", [(1, b"aaa"), (2, None), (3, b"bbb")], [True, False, False]),
+        ("null", None, [(1, 10), (2, None), (3, 20)], [False, True, False]),
+    ],
+)
+def test_null_safe_equals_scalar_both_directions(type_name, test_value, test_data, expected_values):
+    """Test null-safe equality with scalars on both LHS and RHS of the comparison."""
```

Comment:
What's the difference between this test and the previous one besides the null testcase?